GVFS/GVFS.Platform.Windows/WindowsFileSystem.cs

```diff
@@ -122,7 +122,35 @@ public bool IsSocket(string fileName)
             return false;
         }
 
-        public bool TryCreateDirectoryWithAdminOnlyModify(ITracer tracer, string directoryPath, out string error)
+        public bool TryCreateDirectoryWithAdminAndUserModifyPermissions(string directoryPath, out string error)
+        {
+            try
+            {
+                DirectorySecurity directorySecurity = new DirectorySecurity();
+
+                // Protect the access rules from inheritance and remove any inherited rules
+                directorySecurity.SetAccessRuleProtection(isProtected: true, preserveInheritance: false);
+
+                // Add new ACLs for users and admins.  Users will be granted write permissions.
+                WindowsFileSystem.AddUsersAccessRulesToDirectorySecurity(directorySecurity, grantUsersModifyPermissions: true);

```

Comment:
nit: below, we use the unqualified access to `AddUsersAccessRulesToDirectorySecurity` and friends

Reply:
Nice catch!